### ISSUE TITLE
JAX silently discards masks in numpy masked arrays

### ISSUE THREAD

**Issue by jakevdp** [google/jax#13795]
Reported in https://github.com/data-apis/array-api/issues/582
```python
>>> jnp.sin(np.ma.array([1, 2, 3], mask=[True, False, True]))
Array([0.84147096, 0.9092974 , 0.14112   ], dtype=float32)
```
